mav_trajectory_generation/src/rpoly.cpp

```diff
@@ -40,30 +40,66 @@
 
 namespace mav_trajectory_generation {
 
-Eigen::VectorXcd findRootsJenkinsTraub(
-    const Eigen::VectorXd& coefficients_increasing) {
-  const Eigen::VectorXd coefficients_decreasing =
-      coefficients_increasing.reverse();
+int findLastNonZeroCoeff(const Eigen::VectorXd& coefficients) {
+  int last_non_zero_coefficient = -1;
+
+  // Find last non-zero coefficient:
+  for (size_t i = coefficients.size() - 1; i != -1; i--) {
+    if (coefficients(i) != 0.0) {
```

Comment:
Most of the time you compute the roots for the n-th derivative. And the last n coefficients are exactly zero then. Don't know..